22.RaytracedAO/CommandLineHandler.hpp

```diff
@@ -91,6 +108,33 @@ class CommandLineHandler
 				sceneDirectory = rawVariables[REA_SCENE].value();
 			if(rawVariables[REA_TERMINATE].has_value())
 				terminate = true;
+			if (rawVariables[REA_PROCESS_SENSORS].has_value())
+			{
+				processSensorsBehaviour = ProcessSensorsBehaviour::PSB_RENDER_ALL_THEN_INTERACTIVE; // default, when no behaviour option is specified
+
+				const auto& values = rawVariables[REA_PROCESS_SENSORS].value();
+				if (!values.empty())
+				{
+					const char* behaviour = values[0].c_str();
+					if (strcmp(behaviour, "RenderAllThenTerminate") == 0)
+					{
+						processSensorsBehaviour = ProcessSensorsBehaviour::PSB_RENDER_ALL_THEN_TERMINATE;
+						assert(values.size() == 1 && "Passing a sensor ID doens't make sense here.");
+					}
+					else if (strcmp(behaviour, "RenderSensorThenInteractive") == 0)
+					{
+						processSensorsBehaviour = ProcessSensorsBehaviour::PSB_RENDER_SENSOR_THEN_INTERACTIVE;
+						sensorID = std::stoi(values[1]);
+					}
+					else if (strcmp(behaviour, "InteractiveAtSensor") == 0)
+					{
+						processSensorsBehaviour = ProcessSensorsBehaviour::PSB_INTERACTIVE_AT_SENSOR;
+						sensorID = std::stoi(values[1]);
+					}
+					else
+						assert(!"Invalid option.");
```

Comment:
actually log an error and set a sane defaul, asserts (crashes) are not acceptable in this application

Reply:
Resolved.